File: 1-TwoSum/1-TwoSum.js
// Last updated: 4/29/2026, 10:37:01 AM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hash = {};
    for(let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const other = target - num;
        if(hash[other] !== undefined) {
            return [hash[other], i];
        }
        hash[num] = i;
    }
    return [];
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 4/29/2026, 10:37:04 AM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in dict:
                return [dict[complement], i]
            dict[nums[i]] = i
        return []